### PR TITLE
Improved metatag usage

### PR DESCRIPTION
**Pull request by dominiclord**
- Alternate language listing
- Allow hooking into final metatag methods
- Configurable reCaptcha lang

### CODE REVIEW

File: src/Cms/AbstractWebTemplate.php [hunk 1]
```diff
@@ -289,4 +360,93 @@ public function twitterCardImage()
...
+            $origLang   = $this->currentLanguage();
+
+            foreach ($this->availableLanguages() as $lang) {
+                if ($lang === $origLang) {
+                    continue;
+                }
+
+                $this->localesManager->setCurrentLocale($lang);
+
+                $data = [
```
Comment (mcaskill) [inline src/Cms/AbstractWebTemplate.php]: Let's move the array format to a separate method for easy customization:
`protected function formatAlternateTranslation()`

File: src/Cms/AbstractWebTemplate.php [hunk 2]
```diff
@@ -289,4 +360,93 @@ public function twitterCardImage()
 
         return $this->metaImage();
     }
+
+
+
+    // Polylingual
```
Comment (mcaskill) [inline src/Cms/AbstractWebTemplate.php:372]: I propose we move these methods to a `LocaleAwareTrait` except for `AbstractWebTemplate::currentLanguage()` and `AbstractWebTemplate::currentLocale()`.

File: src/Cms/AbstractWebTemplate.php [hunk 3]
```diff
@@ -153,14 +175,31 @@ public function metaDescription()
...
         $context = $this->contextObject();
 
+        $desc = null;
         if ($context instanceof HasMetatagInterface) {
-            return $context['meta_description'];
+            $desc = $context['meta_description'];
+        }
+
+        if (!$desc) {
+            $desc = $this->metaDescriptionFromConfig();
```
Comment (mcaskill) [inline src/Cms/AbstractWebTemplate.php]: Is there a reason the fallback methods are called `meta*FromConfig()`?
I think ``fallbackMeta*()` would be better.

**Review by mcaskill — CHANGES_REQUESTED** [review]

**Comment by dominiclord** [comment]
Not quite sure how to implement your `formatAlternateTranslation` suggestion. Seems like a good idea though.

File: src/Cms/LocaleAwareTrait.php [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+     * after processing all available languages.
+     *
+     * @param  mixed  $context The translated {@see \Charcoal\Model\ModelInterface model}
+     *     or array-accessible structure.
+     * @param  string $lang    The currently iterated language.
+     * @return array Returns a link structure.
+     */
+    protected function formatAlternateTranslation($context, $lang)
+    {
+        $isModel    = ($context instanceof ModelInterface);
```
Comment (mcaskill) [inline src/Cms/LocaleAwareTrait.php]: Is there ever an occasion where we wouldn't be processing a `ModelInterface`?

I defined `ContextualTemplate::contextObject()` mixin to return a model.

**Pull request merged**